src/codeflare_sdk/cluster/auth.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2022 IBM, Red Hat
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+The auth sub-module contains the definitions for the Authentication objects, which represent
+the methods by which a user can authenticate to their cluster(s). The abstract class, `Authentication`,
+contains two required methods `login()` and `logout()`. Users can use one of the existing concrete classes to
+authenticate to their cluster or add their own custom concrete classes here.  
+"""
+
+import abc
+import openshift as oc
+
+
+class Authentication(metaclass=abc.ABCMeta):
+    """
+    An abstract class that defines the necessary methods for authenticating to a remote environment.
+    Specifically, this class defines the need for a `login()` and a `logout()` function.
+    """
+
+    def login(self):
+        pass
+
+    def logout(self):
+        pass
+
+
+class TokenAuthentication(Authentication):
+    """
+    `TokenAuthentication` is a subclass of `Authentication`. It can be used to authenticate to an OpenShift
+    cluster when the user has an API token and the API server address.
+    """
+
+    def __init__(
+        self,
+        token: str = None,
+        server: str = None,
+    ):
+
+        self.token = token
+        self.server = server
+
+    def login(self):
```

Comment:
Could you also add simple function docstrings? To the init, login, and logout function for each implementation.

Reply:
yes, my bad. Should have add those :upside_down_face: